tools/blocksconvert/scheduler/scheduler.go

```diff
@@ -51,19 +54,29 @@ func NewScheduler(cfg Config, scfg blocksconvert.SharedConfig, l log.Logger, reg
 		users = blocksconvert.ParseAllowedUsers(cfg.AllowedUsers)
 	}
 
-	s := newSchedulerWithBucket(l, b, scfg.BucketPrefix, users, cfg, reg)
+	var ignoredUserRegex *regexp.Regexp = nil
+	if users.AllUsersAllowed() && cfg.IgnoredUserPattern != "" {
```

Comment:
Why not applying ignored users even if `users.AllUsersAllowed()` is false?

Reply:
You're right. I copied the logic from scanner, where these two options are exclusive. It works slightly differently here, although I would only recommend using one option.

Reply:
I've modified scanner to have the same logic wrt. to options as scheduler. (Internally the logic was already the same, just config options were treated differently).

Reply:
Ok. Makes sense to me.